packages/core-contracts/src/contracts/arks/SiUSDArk.sol

```diff
@@ -0,0 +1,190 @@
+                            STATE VARIABLES
+    //////////////////////////////////////////////////////////////*/
+
+    /// @notice The InfiniFi Gateway contract for interacting with the protocol
+    IInfiniFiGateway public immutable gateway;
+
+    /// @notice The siUSD (staked iUSD) ERC4626 vault contract
+    IERC4626 public immutable siUSD;
+
+    /*//////////////////////////////////////////////////////////////
+                                ERRORS
+    //////////////////////////////////////////////////////////////*/
+
+    error InvalidGatewayAddress();
+    error InvalidSiUSDAddress();
+
+    /*//////////////////////////////////////////////////////////////
+                            CONSTRUCTOR
+    //////////////////////////////////////////////////////////////*/
+
+    /**
+     * @notice Constructor to set up the SiUSDArk
+     * @param _gateway Address of the InfiniFi Gateway contract
+     * @param _siUSD Address of the siUSD ERC4626 vault
+     * @param _params ArkParams struct containing necessary parameters for Ark initialization
+     */
+    constructor(
+        address _gateway,
+        address _siUSD,
+        ArkParams memory _params
+    ) Ark(_params) {
+        if (_gateway == address(0)) {
+            revert InvalidGatewayAddress();
+        }
+        if (_siUSD == address(0)) {
+            revert InvalidSiUSDAddress();
+        }
+
+        gateway = IInfiniFiGateway(_gateway);
+        siUSD = IERC4626(_siUSD);
+
+        // Approve Gateway to spend USDC for mintAndStake operations
+        config.asset.forceApprove(_gateway, Constants.MAX_UINT256);
+
+        // Approve Gateway to spend siUSD for unstake operations
+        IERC20(_siUSD).forceApprove(_gateway, Constants.MAX_UINT256);
+
+        // Approve Gateway to spend iUSD (siUSD's asset) for redeem operations
+        IERC20(siUSD.asset()).forceApprove(_gateway, Constants.MAX_UINT256);
+    }
+
+    /*//////////////////////////////////////////////////////////////
+                            VIEW FUNCTIONS
+    //////////////////////////////////////////////////////////////*/
+
+    /**
+     * @inheritdoc IArk
+     * @notice Returns the total assets managed by this Ark, denominated in USDC
+     * @dev Converts siUSD shares -> iUSD (via ERC4626)
+     *      Note: iUSD is approximately 1:1 with USDC (adjusted for decimals)
+     *      iUSD has 18 decimals, USDC has 6 decimals
+     * @return assets The total balance of USDC-equivalent assets
+     */
+    function totalAssets() public view override returns (uint256 assets) {
+        uint256 siUSDShares = siUSD.balanceOf(address(this));
+        if (siUSDShares > 0) {
+            // Convert siUSD shares to iUSD amount (18 decimals)
+            uint256 iUSDAmount = siUSD.convertToAssets(siUSDShares);
+            // Convert iUSD (18 decimals) to USDC equivalent (6 decimals)
+            // iUSD is pegged 1:1 to USD, so we just adjust decimals
+            assets = iUSDAmount / 1e12;
```

Comment:
Also rounding, is there an issue with this? This is always rounding towards zero, meaning we will usually report a bit less than we have in the vault. Is this an issue?

Reply:
I think we can accept it. We'll potentially leave some dust in the vault but it won't be meaningful